src/targets/gpu/compile_hip.cpp

```diff
@@ -228,9 +228,15 @@ std::vector<std::vector<char>> compile_hip_src_with_hiprtc(std::vector<hiprtc_sr
 bool hip_has_flags(const std::vector<std::string>& flags)
 {
     hiprtc_program prog{" "};
```

Comment:
This is variable is not used. Also the other `hip_has_flags` function should be removed.